cpp/monitoring/latency.h

```diff
@@ -0,0 +1,137 @@
+#ifndef CERT_TRANS_MONITORING_LATENCY_H_
+#define CERT_TRANS_MONITORING_LATENCY_H_
+
+#include <string>
+
+#include "base/macros.h"
+#include "monitoring/counter.h"
+#include "monitoring/monitoring.h"
+
+namespace cert_trans {
+
+class ScopedLatency;
+
+
+// A helper class for monitoring latency.
+// This class creates two Counter metrics, one called |base_name| which
+// contains the sum of all latencies broken down by labels, and another called
+// "|base_name|_count" which contains the number of latency measurements taken,
+// also broken down by labels.
+//
+// To actually measure latency, you can either call RecordLatency() directly
+// with a latency sample, or use the ScopedLatency() method to return an object
+// which will automatically add a latency measurement consisting of the
+// duration between the call to ScopedLatency() and the destruction of the
+// returned object.
+//
+// The |TimeUnit| template parameter is used to specify the unit of the values
+// added to the counter, e.g. specifying std::chrono::milliseconds will
+// duration_cast all recorded latencies to milliseconds before adding to the
+// total count.
+//
+// |LabelTypes...| works as in the Counter<> and Gauge<> templates.
+//
+// Example usage:
+//
+//   static Latency<std::chrono::milliseconds, std::string> latency_by_name(
+//      "latency_by_name", "name");
+//   ...
+//
+//   void DoStuffForName(const string& name) {
+//     // This will record the latency of this method for each different |name|
+//     ScopedLatency latency(latency_by_name.ScopedLatency(name));
+//     ...
+//     // do stuff
+//     ...
+//   }
+//
+//   // Here's an example where there's no one scope so can't easily use the
+//   // ScopedLatency helper.
+//   void FinishedDoingStuffCallback(const string& name,
+//                                   const system_clock::time_point started_at)
+//                                   {
+//      latency_by_name.RecordLatency(name, system_clock::now() - started_at);
+//   }
+template <class TimeUnit, class... LabelTypes>
+class Latency {
```

Comment:
Yeah, that's what I meant, it remembers what unit the variable is in for you.

I'd say leave it like this for now, when we need more, we can refactor it. If it had been already exactly the way an hypothetical ```EventMetric``` would have been, I'd have pushed to "just rename it", but it isn't. :smiley: